src/org/labkey/test/tests/SampleSetRemoteAPITest.java

```diff
@@ -181,6 +196,328 @@ public void sampleSetWithMissingValueField() throws IOException, CommandExceptio
+        JSONObject sixthProps = new JSONObject();
+        sixthProps.put("volume", 21.5);
+        sixthProps.put("color", "green");
+        sixthMaterial.put("properties", sixthProps);
+        Material material6 = new Material(sixthMaterial);
+
+        JSONObject sampleMaterial7 = new JSONObject();
+        sampleMaterial7.put("name", "7th");
+        sampleMaterial7.put("sampleSet", sampleSet);
+        Material material7 = new Material(sampleMaterial7);
+
+        JSONObject sampleMaterial4 = new JSONObject();
+        sampleMaterial4.put("name", "4th");
+        sampleMaterial4.put("sampleSet", sampleSet);
+        Material material4 = new Material(sampleMaterial4);
+
+        Run runA = new Run();
+        runA.setProperties(Collections.singletonMap("RunFileField", "foo.xls"));
+        Data runAData = new Data();
+        runAData.setName("joe");
+        runAData.setPipelinePath("/");
+        runA.setDataInputs(Arrays.asList(runAData));
+        runA.setName("saveBatchMatrials A");
+        runA.setMaterialOutputs(Arrays.asList(material5, material6));       // 5 and 6 don't exist yet, they should be created as outputs
+
+        Run runB = new Run();
+        runB.setProperties(Collections.singletonMap("RunFileField", "foo.xls"));
+        Data runBData = new Data();
+        runBData.setName("billy");
+        runBData.setPipelinePath("/");
+        runB.setDataInputs(Arrays.asList(runBData));
+        runB.setName("saveBatchMaterials B");
+        runB.setMaterialInputs(Arrays.asList(material4)); // material4 should already exist
+        runB.setMaterialOutputs(Arrays.asList(material7));
+
+        batch.getRuns().add(runA);
+        batch.getRuns().add(runB);
+
+        SaveAssayBatchCommand saveAssayBatchCommand = new SaveAssayBatchCommand(SaveAssayBatchCommand.SAMPLE_DERIVATION_PROTOCOL, batch);
+        //saveAssayBatchCommand.setAssayId(Integer.parseInt(assayIdStringValue));
+        SaveAssayBatchResponse saveAssayBatchResponse;
+        saveAssayBatchResponse = saveAssayBatchCommand.execute(createDefaultConnection(true), getProjectName() + "/" + samplesFolder);
+
+        refresh();
+        // add a bodyWebPart here to make viewing the sampleset easier while debugging
+        new PortalHelper(getDriver()).addBodyWebPart("Sample Sets");
+        refresh();
+        DataRegionTable sampleSetList =  DataRegionTable.DataRegion(getDriver()).withName("SampleSet").waitFor();
+        waitAndClick(Locator.linkWithText(sampleSetName));
+        DataRegionTable materialsList =  DataRegionTable.DataRegion(getDriver()).withName("Material").waitFor();
+
+        // get the samples from the sampleset
+        SelectRowsResponse sampleSetResponse = dgen.getRowsFromServer(createDefaultConnection(true));
+
+        // at this point, the API either doesn't work or requires some hack to make it work.
+        // the materialoutputs are present, but the API does not create samples derived from the runs.
+
```

Comment:
I realize this test is ignored but it looks like the intent of this test is to validate that you can derive a sample but doesn't have anything to do with missing values. Correct?

Reply:
Correct.  I've expanded the comment; this documents the behavior of saveBatch() with regard to the differences between the JavaAPI and the JSAPI.
There's more difference than I'd like to see between the two, but so far that appears to be a 'feature'